src/System.Globalization/tests/Invariant/InvariantMode.cs

```diff
@@ -794,5 +794,22 @@ public void TestHashing()
             string turkishString = "i\u0130";
             Assert.Equal(ordinalComparer.GetHashCode(turkishString), cultureComparer.GetHashCode(turkishString));
         }
+
+        [Theory]
+        [InlineData((int)'a', (int)'A', (int)'a')]
+        [InlineData((int)'A', (int)'A', (int)'a')]
+        [InlineData(0x00C1, 0x00C1, 0x00C1)] // U+00C1 LATIN CAPITAL LETTER A WITH ACUTE
+        [InlineData(0x00E1, 0x00E1, 0x00E1)] // U+00E1 LATIN SMALL LETTER A WITH ACUTE
+        [InlineData(0x00D7, 0x00D7, 0x00D7)] // U+00D7 MULTIPLICATION SIGN
+        public void TestRune(int original, int expectedToUpper, int expectedToLower)
+        {
+            Rune originalRune = new Rune(original);
+
+            Assert.Equal(expectedToUpper, Rune.ToUpperInvariant(originalRune).Value);
+            Assert.Equal(expectedToUpper, Rune.ToUpper(originalRune, CultureInfo.InvariantCulture).Value);
+
+            Assert.Equal(expectedToLower, Rune.ToLowerInvariant(originalRune).Value);
+            Assert.Equal(expectedToLower, Rune.ToLower(originalRune, CultureInfo.InvariantCulture).Value);
```

Comment:
If these tests are meant to run in invariant mode, doesn't that mean that we'll end up effectively applying the invariant culture even if it isn't specified?  And if so, shouldn't the tests for that support then also verify when "Invariant" isn't specified?

Reply:
When we have the invariant mode is on, you can still create cultures like en-US but the underlying data of such cultures will come from the invariant culture data. in such tests, using invariant culture or any other culture should end up with the same result. but it may be a good idea to validate that as @stephentoub suggested.

Reply:
Good feedback - I've addressed this in latest iteration.